### PR TITLE
fix(tomorrow): advance displayed deadline for bumped due-today goals

### PR DESCRIPTION
**Pull request by narthur**
## Summary

After the baremin-bumping fixes landed, a `clean` goal due today at 5:59 PM was reported as:

```
clean   +01:25:00 in 1 day            57m   5:59 PM
```

The baremin (`+01:25:00 in 1 day`) is correct — it's the amount needed by *tomorrow's* deadline. But the timeframe (`57m`) and absolute deadline (`5:59 PM`) still came from the goal's own losedate, i.e. *today's* 5:59 PM. Those columns belong to the bumped baremin's coverage window, not the goal's untouched losedate.

## Fix

- Thread a `losedateFor func(Goal) int64` formatter through `handleFilteredCommandWithDisplay` (sibling of the existing `bareminFor`).
- New `losedateByEndOfTomorrowAt(g, now)` returns `g.Losedate + 86400` for due-today goals and `g.Losedate` otherwise.
- `handleTomorrowCommand` wires the new formatter in. All other views (today, all, do-less, due-within, list) keep the existing pass-through behaviour via `handleFilteredCommand`.

For the `clean` goal this produces:

```
clean   +01:25:00 in 1 day            1d  5:59 PM
```

## Test plan

- [x] New `TestLosedateByEndOfTomorrowAt` covers due-today (+24h bump), due-tomorrow (passthrough), and due-later (passthrough)
- [x] Existing `TestBareminByEndOfTomorrowAt` and `TestRoadallSlopePerDayAt` still pass — no behaviour change for other paths
- [ ] Manually verify against the live `clean` goal after deploy

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Tomorrow view now advances displayed deadlines by one calendar day for goals due today and aligns timeframe/absolute-deadline displays with that projected "tomorrow" moment; rendered order follows the adjusted displayed deadline.

* **Bug Fixes**
  * Overdue goals no longer get their displayed deadline bumped; displayed deadlines respect local calendar/DST boundaries.

* **Tests**
  * Added tests for deadline projection, DST edge cases, overdue behavior, and sorting by displayed deadline.

[Review Change Stack](https://app.coderabbit.ai/change-stack/PinePeakDigital/buzz/pull/240)

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -417,6 +417,52 @@ func piecewiseRoadall(startT int64, startV float64, segments ...float64) [][]*fl
...
+		{
+			name:     "due tomorrow keeps own losedate",
+			goal:     Goal{Losedate: tomorrowDeadline},
+			expected: tomorrowDeadline,
+		},
+		{
+			name:     "due later keeps own losedate",
+			goal:     Goal{Losedate: dayAfterTomorrowDeadline},
+			expected: dayAfterTomorrowDeadline,
+		},
```
[the remark this answers is not in the document]
Comment (narthur) [reply main_test.go:461]: Addressed in b1dce01 — TestLosedateByEndOfTomorrowAt now has an 'overdue keeps own losedate' case and TestBareminByEndOfTomorrowAt has an 'overdue is unchanged' case. Both lock in the dueLaterTodayAt gating predicate.
